haystack/dataclasses/chat_message.py

```diff
@@ -16,6 +18,24 @@ class ChatRole(str, Enum):
     FUNCTION = "function"
 
 
+class ContentType(str, Enum):
+    """Enumeration representing the different types of content that fit in a ChatMessage."""
+
+    TEXT = "text"
+    IMAGE_URL = "image_url"
+    IMAGE_BASE64_JPG = "image_base64/jpg"
+    IMAGE_BASE64_PNG = "image_base64/png"
```

Comment:
Just an update here, I don't think the approach of a different Enum for every different subtype of image is a wise idea. This will end up with dozens of different types in the future. I think a better approach would be to keep some simple content types (e.g. TEXT, IMAGE_URL, IMAGE_BASE64), and then implement a resolver that can map a given ByteStream object into one of the handful of ContentTypes. This resolver can then be called by `ChatMessage._parse_byte_stream_content` to return the correct ContentType (see next comment).